### ISSUE TITLE
[Sourcing] Tree table: New ability to store and recover last table state after editing table data

### ISSUE THREAD

**Issue by yixuan-liu** [SAP/fundamental-ngx#9752]
Is this a bug, enhancement, or feature request?
--
Feature request

Briefly describe your proposal.
--
When we try to edit/update row data in table via inputs on another page/view (we use Angular router to a different page), the last table state does not get persisted but instead gets a total refresh.​

For example, when a new row is dynamically added and table gets refreshed, table does not scroll into that position where the newly added row resides. We'd like to recover to that last table page position after performing any edit to table data.

Whenever user performs any type of editing to the row data on a new page (navigated via Angular router), the table view should retain its last viewed page when user navigates back from that editing page view.

We need a 'scrollTo' strategy so that we can customize which table page the table could display, instead of always showing the 1st table page (with the very first 20 rows or so) upon completing editing.

Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
--
Current version.

**Comment by N1XUS** [comment]
@yixuan-liu Hello. Thank you for submitting this ticket. If you could provide some code example (via stackblitz) to simulate the behaviour you have in your application, that would help a lot with understanding the desired behaviour

**Comment by yixuan-liu** [comment]
Thanks @N1XUS : we will try it out and let you know

**Comment by CSENoni** [comment]
Found issue: https://github.com/SAP/fundamental-ngx/issues/9928

**Comment by yixuan-liu** [comment]
Hi @N1XUS, we have tried the fix and it works fine when we have the table component instance present. One issue, however, is that, when we use router to navigate from this page (where the table resides) to another page (for performing some form inputs collection on table data), and then comes back, the host/parent component gets destroyed and reinitialized by Angular. We have tried adopting the `CanReuseRoute` strategy but it requires a lot of modifications to our routes config and HTML structure.

So, our question is, is there a way for us to restore the old table state even when we navigate back and forth between different pages? 

Thanks!
